src/Utilities/Compiler/Extensions/IMethodSymbolExtensions.cs

```diff
@@ -272,7 +282,12 @@ public static DisposeMethodKind GetDisposeMethodKind(this IMethodSymbol method,
             {
                 if (IsDisposeImplementation(method, iDisposable) ||
                     (Equals(method.ContainingType, iDisposable) &&
-                     method.HasDisposeMethodSignature()))
+                     method.HasDisposeMethodSignature())
+#if NET_ANALYZERS
+                    || (method.ContainingType.IsRefLikeType &&
```

Comment:
Can we extract out a helper `IsDisposeMethodOfRefLikeType(IMethodSymbol)` and use it here, as well as in `IsDisposable` method below as `return type.GetMembers("Dispose").OfType<IMethodSymbol>().Any(IsDisposeMethodOfRefLikeType);`

Reply:
I can extract the method for clarity, but what would be the advantage of using it in `IsDisposable` over `HasDisposeSignatureByConvention`?

Reply:
I'll let you make the final call here as all of these are private methods. I was just providing a suggestion that might avoid duplicated code with `#if` directives.